src/analyzer.ts

```diff
@@ -20,12 +22,16 @@ import {Node, queryAll, predicates, getAttribute} from 'dom5';
 
 import {FileCB, VinylReaderTransform} from './streams';
 import {urlFromPath, pathFromUrl} from './path-transformers';
-import {DocumentDeps, getDependenciesFromDocument, isDependencyExternal}
-  from './get-dependencies-from-document';
 
 const minimatchAll = require('minimatch-all');
 const logger = logging.getLogger('cli.build.analyzer');
 
+export interface DocumentDeps {
```

Comment:
looks like maybe the only place that produces one of these includes all fields. Maybe make them non-optional.